src/providers/tfc/runProvider.ts

```diff
@@ -137,13 +144,44 @@ export class RunTreeDataProvider implements vscode.TreeDataProvider<vscode.TreeI
 
       return items;
     } catch (error) {
-      if (axios.isAxiosError(error) && error.response?.status === 401) {
-        vscode.window.showErrorMessage(`Runs not accessible for workspace ${workspaceId}: ${error}`);
-      } else if (error instanceof Error) {
-        vscode.window.showErrorMessage(error.message);
-      } else if (typeof error === 'string') {
-        vscode.window.showErrorMessage(error);
+      let message = `Failed to list runs in ${workspaceId}: `;
+
+      if (error instanceof ZodiosError) {
+        handleZodiosError(error, message, this.outputChannel, this.reporter);
+        return [];
+      }
+
+      if (axios.isAxiosError(error)) {
+        if (error.response?.status === 401) {
+          handleAuthError();
+          return [];
+        }
+
+        if (error.response?.status === 404) {
+          vscode.window.showWarningMessage(
+            `Workspace ${this.activeWorkspace?.attributes.name} (${workspaceId}) not found, please pick another one`,
```

Comment:
```suggestion
            `Workspace ${this.activeWorkspace.attributes.name} (${workspaceId}) not found, please pick another one`,
```

In line 105, we're checking if `this.activeWorkspace` is set, so it should never be undefined here